File: 2023/day1/part1.rb
file = File.open("input.txt")
lines = file.read.split
claibration = 0
lines.each do |line|
    numbers = line.tr('^0-9', '')
    coordinate = "#{numbers[0]}#{numbers[-1]}"
    claibration += coordinate.to_i
end
pp claibration

File: 2024/day1/part1.go
package day1_part1

import (
	"adventofcode/utils"
	"fmt"
	"regexp"
	"strconv"
)

// Executes the puzzle for all days
func Part1(file string) int {
	lines := utils.ReadFileByLine("1_1.txt")
	calibration := 0
	for _, line := range lines {
		rgx := regexp.MustCompile(`\d`)
		numbers := rgx.FindAllString(line, -1)
		coordinate, _ := strconv.Atoi(numbers[0] + numbers[len(numbers)-1])
		calibration += coordinate

	}
	fmt.Println(calibration)
	return calibration
}
